lib/plugins/aws/deploy/compile/events/stream/index.js

```diff
@@ -78,8 +88,15 @@ class AwsCompileStreamEvents {
                 .Error(errorMessage);
             }
 
-            const streamType = EventSourceArn.split(':')[2];
-            const streamName = EventSourceArn.split('/')[1];
+            const streamType = event.stream.type || EventSourceArn.split(':')[2];
+            const streamName = (function () {
+              if (EventSourceArn['Fn::GetAtt']) {
+                return EventSourceArn['Fn::GetAtt'][0];
+              } else if (EventSourceArn['Fn::ImportValue']) {
+                return EventSourceArn['Fn::ImportValue'];
```

Comment:
Given that this is used to generate (line 103) the logical ID of the stream object (used on line 140).  I think there is a risk that long identifiers for exported values could be too long according to a logical ID name length limit enforced by CloudFormation ("Resource name - Maximum size of a resource name. - 255 characters" - from http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html).
This is tricky because framework users will not always have control over that.